xla/service/gpu/matmul_utils.cc

```diff
@@ -636,25 +626,21 @@ absl::Status RunGemm(const GemmConfig& config, se::DeviceMemoryBase lhs_buffer,
                      std::optional<se::blas::AlgorithmType> algorithm,
                      se::blas::ProfileResult* profile_result) {
   VLOG(2) << "Executing a GemmThunk";
-
   TF_ASSIGN_OR_RETURN(
       GemmConfig::DescriptorsTuple desc,
       config.GetMatrixDescriptors(lhs_buffer, rhs_buffer, output_buffer));
 
   se::NumericOptions numeric_options{
       deterministic_ops,
-      /*allow_tf32=*/config.compute_precision <= 1};
+      /*allow_tf32=*/config.compute_precision <= 1,
+      config.grad_flags};
 
   if (!algorithm) algorithm = config.algorithm;
 
-  se::blas::CallContext context = se::blas::CallContext::kNone;
-  if (config.grad_x) {
-    context = desc.operands_swapped ? se::blas::CallContext::kBackpropInput2
-                                    : se::blas::CallContext::kBackpropInput1;
-  }
-  if (config.grad_y) {
-    context = desc.operands_swapped ? se::blas::CallContext::kBackpropInput1
-                                    : se::blas::CallContext::kBackpropInput2;
+  if(desc.operands_swapped) {
+    numeric_options.grad_flags = ((numeric_options.grad_flags & 1)<<1)
```

Comment:
This is swapping the lowest two bits, right? Please pull out into a function with a descriptive name or add a comment.